pkg/networkdevice/metadata/payload_utils_test.go

```diff
@@ -42,7 +43,7 @@ func Test_batchPayloads(t *testing.T) {
 			FlowType:  "netflow5",
 		})
 	}
-	payloads := BatchPayloads("my-ns", "127.0.0.0/30", collectTime, 100, devices, interfaces, ipAddresses, topologyLinks, netflowExporters)
+	payloads := BatchPayloads("my-ns", "127.0.0.0/30", collectTime, 100, devices, interfaces, ipAddresses, topologyLinks, netflowExporters, nil)
```

Comment:
Can we test diagnoses similar to other resources here: https://github.com/DataDog/datadog-agent/blob/2dbe70998beb435b2a57b1060226509d2be80782/pkg/networkdevice/metadata/payload_utils_test.go#L32-L45

to verify that the batching works as expected.

Reply:
Added the test
In theory in this first iteration, only zero or one diagnoses payload is sent with each device